### ISSUE TITLE
A sensor extension

### ISSUE THREAD

**Issue by DNin01** [TurboWarp/extensions#1431]
Sometimes you want to make games or interactive elements that respond to input other than mouse, keyboard, or touch, and for this, it could be really cool to have an extension that can read various sensors of the device, such as the accelerometer or gyroscope.

We could implement anything that is offered by the [Sensor APIs](https://developer.mozilla.org/en-US/docs/Web/API/Sensor_APIs).

Sensing Plus has a block to get accelerometer/gyroscope data but it also has a bunch of other blocks that I don't necessarily want to have to import along. Plus, a dedicated extension would probably be programmed better as sensor access would be the whole point of it.

**Comment by DNin01** [comment]
We could also use [device orientation events](https://developer.mozilla.org/en-US/docs/Web/API/Device_orientation_events), which are more widely supported.

Keep in mind that the behavior may differ between browsers.

Note that Safari for macOS supports these APIs since version 17, but Safari for iOS has supported them since version 4.2, released in 2010. On that note, seeing that people are probably going to be making mostly mobile games in TurboWarp that utilize these features, we may not even want a motion sensing extension to work on computers.

**Comment by DNin01** [comment]
Some of you developers might like to try https://sensor-js.xyz/demo.html on your mobile devices to test out the features.

**Comment by Xeltalliv** [comment]
I made this https://github.com/Xeltalliv/extensions/blob/generic-sensors/extensions/Vadik1/generic-sensors.js over a year ago, but haven't submitted to any extension gallery

**Comment by JuanJose2325** [comment]
Hi, I don't know if I should post this here, but I'm having trouble getting the values for the gyroscope and accelerometer sensors with the sensing+ extension and the generic sensors extension that was posted here.
However, in https://sensor-js.xyz/demo.html I am getting the values, I have already tested on two phones.

**Comment by DNin01** [comment]
> Hi, I don't know if I should post this here, but I'm having trouble getting the values for the gyroscope and accelerometer sensors with the sensing+ extension and the generic sensors extension that was posted here. However, in https://sensor-js.xyz/demo.html I am getting the values, I have already tested on two phones.

Yeah, Sensing+ implements it in a finicky way, I guess. That's the nature of trying to do so many things in the same extension...

**Comment by SharkPool-SP** [comment]
Closing this as Sensing+ exists and the blocks for it (that were previously buggy) were fixed. If you dont like having the other blocks imported with it, make an extension for sensors. We're happy to accept that.
